spec/features/course/unread_status_management_spec.rb

```diff
@@ -0,0 +1,81 @@
+require 'rails_helper'
+
+RSpec.describe 'Announcements read/unread status management', type: :feature do
+
+  subject { page }
+
+  let!(:instance) { create(:instance) }
+
+  with_tenant(:instance) do
+
+    let(:user) { create(:user, role: :administrator) }
+    let(:course) { create(:course) }
+
+    before do
+      login_as(user, scope: :user)
+    end
+
+    context 'there is a new announcement' do
+      let!(:announcement) { create(:course_announcement, course: course) }
+
+      it 'unread announcements number should increase by 1' do
+        visit course_achievements_path(course)
+        expect(page).to have_link('Announcements (1)')
+      end
+    end
+
+    context 'there are five new announcements' do
+      let!(:announcements) { create_list(:course_announcement, 5, course: course) }
+
+      it 'unread number should increase by 5' do
+        visit course_achievements_path(course)
+        expect(page).to have_link('Announcements (5)')
+      end
+    end
+
+    context 'visit announcements index page' do
+      let!(:announcements) { create_list(:course_announcement, 5, course: course) }
+
+      it 'unread announcements number should be 0' do
+        visit course_announcements_path(course)
+        expect(page).to have_link('Announcements (0)')
+      end
+    end
+
+    context 'visit a certain announcement' do
+      let!(:announcement_1) { create(:course_announcement, course: course) }
+      let!(:announcement_2) { create(:course_announcement, course: course) }
+      let!(:announcement_3) { create(:course_announcement, course: course) }
+
+      it 'unread announcements number should decrease by 1' do
+        visit course_announcement_path(course, announcement_2)
+        expect(page).to have_link('Announcements (2)')
+      end
+    end
+
+    context 'create an announcement for one course and visit another course page' do
+      let!(:announcement) { create(:course_announcement, course: course) }
+      let!(:course_2) { create(:course) }
+
+      it 'unread announcements number of another course will not change' do
+        visit course_achievements_path(course_2)
+        expect(page).to have_link('Announcements (0)')
+      end
+    end
+
+    context 'user create an announcement for a course himself' do
+      let(:announcement) { build(:course_announcement, course: course) }
+
+      before do
+        visit new_course_announcement_path(course)
+        fill_in 'course_announcement_title',    with: announcement.title
```

Comment:
Unnecessary spacing detected.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.